ampe.c

```diff
@@ -429,8 +430,9 @@ static int check_frame_protection(struct candidate *cand, struct ieee80211_mgmt_
     ampe_ie_len = len -
                 (elems->mic + elems->mic_len - (unsigned char *)mgmt);
 
-    /* expect at least MGTK + RSC + expiry */
-    if (ampe_ie_len < 2 + sizeof(struct ampe_ie) + 16 + 8 + 4) {
+    /* expect at least MGTK + RSC + expiry for open/confirm */
+    if (ftype != PLINK_CLOSE &&
```

Comment:
Yes, but keys (GTKData/IGTKData) are not there for close frames (by omission in 13.5.5.4.1).  There wouldn't really be any point to sending it.